awx/main/models/inventory.py

```diff
@@ -1404,5 +1405,10 @@ def get_absolute_url(self, request=None):
         return reverse('api:inventory_script_detail', kwargs={'pk': self.pk}, request=request)
 
 
-for cls in PluginFileInjector.__subclasses__():
-    InventorySourceOptions.injectors[cls.__name__] = cls
+awx_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory')}
+supported_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory.supported')}
+entry_points = awx_entry_points if detect_server_product_name() == 'AWX' else {**awx_entry_points, **supported_entry_points}
```

Comment:
How about
```suggestion
def _load_all_entry_points_for(entry_point_groups: list[str], /) -> dict[str, EntryPoint]:
    return {
        ep.name: ep
        for entry_point_group in entry_point_groups
        for ep in entry_points(group=entry_point_group)
    }

is_awx = detect_server_product_name() == 'AWX'
extra_entry_point_groups = () if is_awx else ('inventory.supported', )
entry_points = _load_all_entry_points_for(['inventory', *extra_entry_point_groups])
```
?

Reply:
Although, it probably makes sense to put `_load_all_entry_points_for()` into `awx_plugins.interfaces`, making it reusable.

cc @chrismeyersfsu

Reply:
I lean towards not putting it in `awx_plugins.interfaces` right now. AWX is the only thing we plan on consuming these entry points at this time.